vault/resource_gcp_secret_static_account.go

```diff
@@ -0,0 +1,263 @@
+	}
+
+	log.Printf("[DEBUG] Reading GCP Secrets backend static account %q", path)
+	resp, err := client.Logical().Read(path)
+	if err != nil {
+		return fmt.Errorf("error reading GCP Secrets backend static account %q: %s", path, err)
+	}
+
+	log.Printf("[DEBUG] Read GCP Secrets backend static account %q", path)
+	if resp == nil {
+		log.Printf("[WARN] GCP Secrets backend static account %q not found, removing from state", path)
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("backend", backend)
+	d.Set("static_account", staticAccount)
+
+	for _, k := range []string{"secret_type", "token_scopes", "service_account_email", "service_account_project"} {
+		v, ok := resp.Data[k]
+		if ok {
+			if err := d.Set(k, v); err != nil {
+				return fmt.Errorf("error reading %s for GCP Secrets backend static account %q: %q", k, path, err)
+			}
+		}
+	}
+
+	var binding interface{}
+	if v, ok := resp.Data["bindings"]; ok && v != "" {
+		binding = gcpSecretFlattenBinding(v)
+	} else {
+		binding = gcpSecretFlattenBinding(nil)
+	}
+	if err := d.Set("binding", binding); err != nil {
+		return fmt.Errorf("error reading %s for GCP Secrets backend static account %q", "binding", path)
+	}
+
+	return nil
+}
+
+func gcpSecretStaticAccountUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+	path := d.Id()
+
+	data := map[string]interface{}{}
+	gcpSecretStaticAccountUpdateFields(d, data)
+
+	log.Printf("[DEBUG] Updating GCP Secrets backend static account %q", path)
+
+	_, err := client.Logical().Write(path, data)
+	if err != nil {
+		return fmt.Errorf("Error updating GCP Secrets backend static account %q: %s", path, err)
+	}
+	log.Printf("[DEBUG] Updated GCP Secrets backend static account %q", path)
+
+	return gcpSecretStaticAccountRead(d, meta)
+}
+
+func gcpSecretStaticAccountDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+	path := d.Id()
+
+	log.Printf("[DEBUG] Deleting GCP secrets backend static account %q", path)
+	_, err := client.Logical().Delete(path)
+	if err != nil {
+		return fmt.Errorf("Error deleting GCP secrets backend static account %q", path)
+	}
+	log.Printf("[DEBUG] Deleted GCP secrets backend static account %q", path)
+
+	return nil
+}
+
+func gcpSecretStaticAccountUpdateFields(d *schema.ResourceData, data map[string]interface{}) {
+	if v, ok := d.GetOk("service_account_email"); ok {
+		data["service_account_email"] = v.(string)
+	}
+
+	if v, ok := d.GetOk("secret_type"); ok {
+		data["secret_type"] = v.(string)
+	}
+
+	if v, ok := d.GetOk("token_scopes"); ok && d.Get("secret_type").(string) == "access_token" {
+		data["token_scopes"] = v.(*schema.Set).List()
+	}
+
+	if v, ok := d.GetOk("binding"); ok {
+		bindingsHCL := gcpSecretRenderBindingsFromData(v)
+		log.Printf("[DEBUG] Rendered GCP Secrets backend static account bindings HCL:\n%s", bindingsHCL)
+		data["bindings"] = bindingsHCL
+	} else {
+		data["bindings"] = ""
+	}
```

Comment:
Will the addition of this else block result in different behavior wrt `bindings` for static accounts and rolesets? I see that it's absent for rolesets ([resource_gcp_secret_roleset.go#L267](https://github.com/hashicorp/terraform-provider-vault/blob/master/vault/resource_gcp_secret_roleset.go#L267)).

Reply:
`bindings` is required for rolesets but optional for static accounts. Thus the setting to `""`.

Reply:
Ah, right. Thanks!